### PR TITLE
Add status subresource to HorizontalPodAutoscaler

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Fixes https://github.com/kubernetes/kubernetes/issues/14520

More unit testing is needed in areas of HPA that were never there will need to come in a follow-on.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Flaky test fix for Shippable: https://github.com/kubernetes/kubernetes/pull/14578

**Comment by derekwaynecarr** [comment]
Rebased

File: pkg/client/unversioned/horizontalpodautoscaler.go [hunk 1]
```diff
@@ -94,6 +95,13 @@ func (c *horizontalPodAutoscalers) Update(horizontalPodAutoscaler *experimental.
 	return
```
Comment (erictune) [inline pkg/client/unversioned/horizontalpodautoscaler.go]: FYI, at line 77 above, you may need to do 
```
body, err := api.Scheme.EncodeToVersion(options, latest.GroupOrDie("").GroupVersion)
```
xref: https://github.com/kubernetes/kubernetes/pull/14566/files#r40605889

No need to do in this PR, but could help if you are debugging issues.

**Comment by erictune** [comment]
LGTM

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Comment by derekwaynecarr** [comment]
Rebased.  Will most likely need a new LGTM.

**Comment by derekwaynecarr** [comment]
Rebased, will need a new LGTM (the rebase change was inconsequential)

**Comment by erictune** [comment]
lgtm

**Comment by piosz** [comment]
@derekwaynecarr could you please rebase?

**Comment by derekwaynecarr** [comment]
@piosz - rebased

**Comment by jszczepkowski** [comment]
@derekwaynecarr Great thanks for doing this! Unfortunately, another rebase is needed.

**Comment by derekwaynecarr** [comment]
@k8s-bot test this

**Comment by derekwaynecarr** [comment]
@bgrant0607 @lavalamp - can one of you give a pass at this now that we are passing again after the latest rebase if @erictune is out?

**Comment by bgrant0607** [comment]
I'll take it.

File: pkg/client/unversioned/horizontalpodautoscaler_test.go [hunk 2]
```diff
@@ -120,6 +120,23 @@ func TestHorizontalPodAutoscalerUpdate(t *testing.T) {
...
+			Name:            "abc",
+			Namespace:       ns,
+			ResourceVersion: "1",
+		},
+	}
+	c := &testClient{
+		Request:  testRequest{Method: "PUT", Path: testapi.Extensions.ResourcePath(getHorizontalPodAutoscalersResoureName(), ns, "abc") + "/status", Query: buildQueryValues(nil)},
+		Response: Response{StatusCode: 200, Body: horizontalPodAutoscaler},
+	}
+	response, err := c.Setup(t).Experimental().HorizontalPodAutoscalers(ns).UpdateStatus(horizontalPodAutoscaler)
```
Comment (bgrant0607) [inline pkg/client/unversioned/horizontalpodautoscaler_test.go]: The client is still Experimental()?

**Comment by bgrant0607** [comment]
LGTM

File: pkg/client/unversioned/horizontalpodautoscaler_test.go (hunk 2 above)
Comment (derekwaynecarr) [inline pkg/client/unversioned/horizontalpodautoscaler_test.go]: Yes, I mentioned that to @ncdc just after our chat.  The receiver was not renamed in the move from experiemental->extensions.

**Pull request merged**